### PR TITLE
Add CL2 test command to scalability scenario

### PR DESCRIPTION
**Pull request by prateekgogia**

### CODE REVIEW

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 1]
```diff
@@ -126,9 +126,20 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
   --create-args="${create_args[*]}" \
   --control-plane-size="${KOPS_CONTROL_PLANE_SIZE:-1}"
 
+KUBECONFIG=$(mktemp -t kubeconfig)
```
Comment (dims) [inline tests/e2e/scenarios/scalability/run-test.sh:129]: could you add the `.XXXXX` suffix like the other usages?
https://cs.k8s.io/?q=mktemp%20-t&i=nope&files=&excludeFiles=&repos=

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 2]
```diff
@@ -126,9 +126,20 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
+kops export kubecfg --admin --kubeconfig="${KUBECONFIG}"
+
 if [[ "${RUN_CL2_TEST:-}" == "true" ]]; then
-  # TODO
-  echo "Run CL2 tests"
+  kubetest2 kops "${KUBETEST2_ARGS[@]}" \
+  --test=clusterloader2 \
+  --kubernetes-version="${K8S_VERSION}" \
+  -- \
+  --alsologtostderr \
```
Comment (dims) [inline tests/e2e/scenarios/scalability/run-test.sh:138]: Do we need this? `alsologtostderr`

Comment (prateekgogia) [reply tests/e2e/scenarios/scalability/run-test.sh]: Thanks yeah not required since we are not logging to a file, I have removed it.

**Comment by prateekgogia** [comment]
@hakman @dims PTAL, https://github.com/kubernetes/test-infra/pull/29888 is merged.

**Comment by hakman** [comment]
Tests were merged, but `presubmit-kops-aws-scale-amazonvpc-using-cl2` is failing. Not sure if the fix can be done here though.
/hold cancel

**Comment by prateekgogia** [comment]
Seems like tests are failing because of `SchedulingThroughput error`-
```
E0622 16:11:07.029071   11921 clusterloader.go:249] Test Finished
E0622 16:11:07.029075   11921 clusterloader.go:250]   Test: /home/prow/go/src/k8s.io/perf-tests/clusterloader2/testing/load/config.yaml
E0622 16:11:07.029080   11921 clusterloader.go:251]   Status: Fail
E0622 16:11:07.029084   11921 clusterloader.go:253]   Errors: [measurement call TestMetrics - TestMetrics error: [action start failed for SchedulingMetrics measurement: unexpected error (code: 0) in ssh connection to master: &errors.errorString{s:"error getting SSH client to ubuntu@:22: 'dial tcp :22: connect: connection refused'"}]
measurement call SchedulingThroughput - SchedulingThroughput error: scheduler throughput: actual throughput 33.000000 lower than threshold 100.000000
measurement call TestMetrics - TestMetrics error: [action gather failed for SchedulingMetrics measurement: unexpected error (code: 0) in ssh connection to master: &errors.errorString{s:"error getting SSH client to ubuntu@:22: 'dial tcp :22: connect: connection refused'"} 
```

Looking into whats the KOPS scheduler qps set be default.

**Comment by prateekgogia** [comment]
scheduler qps and burst values are being set as default-
```
I0622 18:17:05.659560      11 flags.go:64] FLAG: --kube-api-burst="100"
I0622 18:17:05.659567      11 flags.go:64] FLAG: --kube-api-qps="50"
```

**Comment by hakman** [comment]
@prateekgogia Looks to me that the missing part is a `MasterIps` and trying to connect to `ubuntu@:22` after that:
```
E0625 05:00:32.538578   11870 clusterloader.go:171] Getting master name error: master node not found
E0625 05:00:32.658860   11870 clusterloader.go:180] Getting master external ip error: didn't find any ExternalIP master IPs
E0625 05:00:32.802670   11870 clusterloader.go:189] Getting master internal ip error: didn't find any InternalIP master IPs
I0625 05:00:32.802714   11870 clusterloader.go:291] Using config: {ClusterConfig:{KubeConfigPath:/tmp/kubeconfig.nBexleb39 RunFromCluster:false Nodes:102 Provider:0xc0006d18f0 EtcdCertificatePath:/etc/srv/kubernetes/pki/etcd-apiserver-server.crt EtcdKeyPath:/etc/srv/kubernetes/pki/etcd-apiserver-server.key EtcdInsecurePort:2382 MasterIPs:[] MasterInternalIPs:[] MasterName: DeleteStaleNamespaces:false DeleteAutomanagedNamespaces:true APIServerPprofByClientEnabled:true KubeletPort:10250 K8SClientsNumber:2 SkipClusterVerification:false} ReportDir:/logs/artifacts ExecServiceConfig:{Enable:true} ModifierConfig:{OverwriteTestConfig:[] SkipSteps:[]} PrometheusConfig:{TearDownServer:true EnableServer:false EnablePushgateway:false ScrapeEtcd:false ScrapeNodeExporter:false ScrapeWindowsNodeExporter:false ScrapeKubelets:false ScrapeMasterKubelets:false ScrapeKubeProxy:true KubeProxySelectorKey:component ScrapeKubeStateMetrics:false ScrapeMetricsServerMetrics:false ScrapeNodeLocalDNS:false ScrapeAnet:false ScrapeCiliumOperator:false ScrapeMastersWithPublicIPs:false APIServerScrapePort:443 SnapshotProject: ManifestPath: StorageClassProvisioner:kubernetes.io/gce-pd StorageClassVolumeType:pd-ssd PVCStorageClass:ssd ReadyTimeout:15m0s PrometheusMemoryRequest:10Gi} OverridePaths:[]}
...
W0625 05:00:45.055612   11870 etcd_metrics.go:78] ETCD measurements will be disabled due to no MasterIps: []
error dialing ubuntu@:22: 'dial tcp :22: connect: connection refused', retrying
```

**Comment by prateekgogia** [comment]
> @prateekgogia Looks to me that the missing part is a `MasterIps` and trying to connect to `ubuntu@:22` after that:
> 
> ```
> E0625 05:00:32.538578   11870 clusterloader.go:171] Getting master name error: master node not found
> E0625 05:00:32.658860   11870 clusterloader.go:180] Getting master external ip error: didn't find any ExternalIP master IPs
> E0625 05:00:32.802670   11870 clusterloader.go:189] Getting master internal ip error: didn't find any InternalIP master IPs
> I0625 05:00:32.802714   11870 clusterloader.go:291] Using config: {ClusterConfig:{KubeConfigPath:/tmp/kubeconfig.nBexleb39 RunFromCluster:false Nodes:102 Provider:0xc0006d18f0 EtcdCertificatePath:/etc/srv/kubernetes/pki/etcd-apiserver-server.crt EtcdKeyPath:/etc/srv/kubernetes/pki/etcd-apiserver-server.key EtcdInsecurePort:2382 MasterIPs:[] MasterInternalIPs:[] MasterName: DeleteStaleNamespaces:false DeleteAutomanagedNamespaces:true APIServerPprofByClientEnabled:true KubeletPort:10250 K8SClientsNumber:2 SkipClusterVerification:false} ReportDir:/logs/artifacts ExecServiceConfig:{Enable:true} ModifierConfig:{OverwriteTestConfig:[] SkipSteps:[]} PrometheusConfig:{TearDownServer:true EnableServer:false EnablePushgateway:false ScrapeEtcd:false ScrapeNodeExporter:false ScrapeWindowsNodeExporter:false ScrapeKubelets:false ScrapeMasterKubelets:false ScrapeKubeProxy:true KubeProxySelectorKey:component ScrapeKubeStateMetrics:false ScrapeMetricsServerMetrics:false ScrapeNodeLocalDNS:false ScrapeAnet:false ScrapeCiliumOperator:false ScrapeMastersWithPublicIPs:false APIServerScrapePort:443 SnapshotProject: ManifestPath: StorageClassProvisioner:kubernetes.io/gce-pd StorageClassVolumeType:pd-ssd PVCStorageClass:ssd ReadyTimeout:15m0s PrometheusMemoryRequest:10Gi} OverridePaths:[]}
> ...
> W0625 05:00:45.055612   11870 etcd_metrics.go:78] ETCD measurements will be disabled due to no MasterIps: []
> error dialing ubuntu@:22: 'dial tcp :22: connect: connection refused', retrying
> ```

Thanks @hakman, looks CL2 is not able to identify the right control plane node from the nodes list. I have added this PR, hopefully this should help with this error - https://github.com/kubernetes/perf-tests/pull/2293

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 3]
```diff
@@ -126,9 +126,22 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
+KUBECONFIG=$(mktemp -t kubeconfig.XXXXXXXXX)
+kops export kubecfg --admin --kubeconfig="${KUBECONFIG}"
+
 if [[ "${RUN_CL2_TEST:-}" == "true" ]]; then
-  # TODO
-  echo "Run CL2 tests"
+  # CL2 uses AWS_SSH_KEY path to ssh to instances for scraping metrics
+  export AWS_SSH_KEY=/root/.ssh/id_ecdsa
+  ssh-keygen -N '' -t ecdsa -b 521 -f "${AWS_SSH_KEY}" <<<y >/dev/null 2>&1
+  KUBETEST2_ARGS+=("--ssh-public-key=${AWS_SSH_KEY}.pub")
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:136]:
```suggestion
  export AWS_SSH_KEY="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 4]
```diff
@@ -126,9 +126,20 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
   --control-plane-size="${KOPS_CONTROL_PLANE_SIZE:-1}"
 
+KUBECONFIG=$(mktemp -t kubeconfig.XXXXXXXXX)
+kops export kubecfg --admin --kubeconfig="${KUBECONFIG}"
+
 if [[ "${RUN_CL2_TEST:-}" == "true" ]]; then
-  # TODO
-  echo "Run CL2 tests"
+  # CL2 uses AWS_SSH_KEY path to ssh to instances for scraping metrics
+  export AWS_SSH_KEY="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:134]:
```suggestion
  export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 5]
```diff
@@ -126,9 +126,20 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
   --control-plane-size="${KOPS_CONTROL_PLANE_SIZE:-1}"
 
+KUBECONFIG=$(mktemp -t kubeconfig.XXXXXXXXX)
+kops export kubecfg --admin --kubeconfig="${KUBECONFIG}"
+
 if [[ "${RUN_CL2_TEST:-}" == "true" ]]; then
-  # TODO
-  echo "Run CL2 tests"
+  # CL2 uses AWS_SSH_KEY path to ssh to instances for scraping metrics
+  export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:134]:
```suggestion
  export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
  export AWS_SSH_KEY="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
  env
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 6]
```diff
@@ -126,9 +126,22 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
+  export AWS_SSH_KEY="/tmp/kops/${CLUSTER_NAME}/id_ed25519.pub"
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:135]:
```suggestion
  ls -l "/tmp/kops/${CLUSTER_NAME}/
  export KUBE_SSH_KEY_PATH="/tmp/kops/${CLUSTER_NAME}/id_ed25519"
  export AWS_SSH_KEY="/tmp/kops/${CLUSTER_NAME}/id_ed25519"
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 7]
```diff
@@ -126,9 +126,23 @@ kubetest2 kops "${KUBETEST2_ARGS[@]}" \
...
   --control-plane-size="${KOPS_CONTROL_PLANE_SIZE:-1}"
 
+KUBECONFIG=$(mktemp -t kubeconfig.XXXXXXXXX)
+kops export kubecfg --admin --kubeconfig="${KUBECONFIG}"
+
 if [[ "${RUN_CL2_TEST:-}" == "true" ]]; then
-  # TODO
-  echo "Run CL2 tests"
+  # CL2 uses AWS_SSH_KEY path to ssh to instances for scraping metrics
+  ls -l "/tmp/kops/${CLUSTER_NAME}/
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:134]:
```suggestion
  ls -l "/tmp/kops/${CLUSTER_NAME}/"
```

**Review by hakman — CHANGES_REQUESTED** [review]

**Comment by hakman** [comment]
Great progress. Thanks @prateekgogia!
Please keep in mind that this still runs on Calico and not AmazonVPC.
/lgtm
/approve

**Pull request merged**